osu.Game/Beatmaps/LocalCachedBeatmapMetadataSource.cs

```diff
@@ -143,15 +174,22 @@ private void prepareLocalCache()
                 {
                     using (var stream = File.OpenRead(cacheDownloadRequest.Filename))
                     using (var outStream = File.OpenWrite(cacheFilePath))
-                    using (var bz2 = new BZip2Stream(stream, CompressionMode.Decompress, false))
-                        bz2.CopyTo(outStream);
+                    {
+                        // ensure to clobber any and all existing data to avoid accidental corruption.
+                        outStream.SetLength(0);
+
+                        using (var bz2 = new BZip2Stream(stream, CompressionMode.Decompress, false))
+                            bz2.CopyTo(outStream);
```

Comment:
This is probably going to be the most contentious here (attempting to replace the cache file in situ). I did it this way because it seemed simpler. I guess I could try writing to yet another file and doing a file swapsie instead at the end but that seemed like it'd be much more complicated for very little gain because IO can and will go wrong anywhere and always...?

Reply:
I'm up for trying this, it's not conventionally how I'd do it but I'm interested to see how it plays out.

Seems to work well on macOS.